components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/AnalyticsMetricsHandler.java

```diff
@@ -82,19 +85,28 @@ public boolean handleResponseOutFlow(MessageContext messageContext) {
         if (messageContext.getPropertyKeySet().contains(InboundWebsocketConstants.WEBSOCKET_SUBSCRIBER_PATH)) {
             return true;
         }
+        messageContext.setProperty(TRANSPORT_HEADERS, ((Axis2MessageContext) messageContext).getAxis2MessageContext().
+                getProperty(TRANSPORT_HEADERS));
+
         AnalyticsDataProvider provider;
         Object skipPublishMetrics = messageContext.getProperty(Constants.SKIP_DEFAULT_METRICS_PUBLISHING);
-        if (skipPublishMetrics != null && (Boolean) skipPublishMetrics) {
-            provider = new AsyncAnalyticsDataProvider(messageContext);
+        if (messageContext.getProperty(APIConstants.API_TYPE) == "GRAPHQL"){
```

Comment:
Swap the check

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/12441/commits/226c49a261240dc5305f96d69aadb6f86edfae84